pkg/agents/AgentTypes/agent2/agent2.go

```diff
@@ -14,11 +14,11 @@ func (a2 *Agent2) Activity() {
 	a2.BaseAgent.Activity()
 }
 
-func GetAgent() *Agent2 {
+func GetAgent(name string) *Agent2 {
 	ag := &Agent2{
 		baseagent.GetAgent(),
 	}
-	ag.SetName("A2")
+	ag.SetName(name)
```

Comment:
You made the change to the A1 class, but not this one. Why?

Reply:
I forgot.